balboa-kafka/src/main/scala/com/socrata/balboa/kafka/consumer/Consumer.scala

```diff
@@ -0,0 +1,74 @@
+package com.socrata.balboa.kafka.consumer
+
+import java.io.IOException
+
+import com.socrata.balboa.metrics.data.DataStore
+import com.socrata.balboa.metrics.impl.JsonMessage
+import kafka.consumer.{ConsumerIterator, KafkaStream}
+
+/**
+ * Created by Michael Hotan on 2/18/15.
+ *
+ * Balboa Kafka Consumer
+ */
+sealed abstract class BalboaConsumer(stream: KafkaStream[Array[Byte], Array[Byte]]) extends Runnable {
+
+  override def run(): Unit = {
+    val it: ConsumerIterator[Array[Byte], Array[Byte]] = stream.iterator()
+    while (it.hasNext())
```

Comment:
Yeah. There's a good argument for integrating some messages from BalboaFastFailCheck into this. In the JMS version we actually stop/restart the AMQ listeners; but we may be able to do something simpler here; maybe just stop calling .next until the failure resolves itself. You would need to sleep a bit if you are in a failure mode so you don't just have a busy loop.